fuse/src/main/java/tachyon/fuse/TachyonFuseFs.java

```diff
@@ -0,0 +1,678 @@
+        mNextOpenFileId += 1;
+      }
+      LOG.debug("{} created and opened in O_WRONLY mode", path);
+
+    } catch (FileAlreadyExistsException e) {
+      LOG.debug("File {} already exists", turi, e);
+      return -ErrorCodes.EEXIST();
+    } catch (IOException e) {
+      LOG.error("IOException on {}", path, e);
+      return -ErrorCodes.EIO();
+    } catch (TachyonException e) {
+      LOG.error("TachyonException on {}", path, e);
+      return -ErrorCodes.EFAULT();
+    } catch (Throwable e) {
+      LOG.error("Unexpected exception on {}", path, e);
+      return -ErrorCodes.EFAULT();
+    }
+
+    return 0;
+  }
+
+  /**
+   * Flushes cached data on Tachyon.
+   *
+   * Called on explicit sync() operation or at close().
+   * @param path The path on the FS of the file to close
+   * @param fi FileInfo data struct kept by FUSE
+   * @return 0 on success, a negative value on error
+   */
+  @Override
+  public int flush(String path, FuseFileInfo fi) {
+    LOG.trace("flush({})", path);
+    final long fd = fi.fh.get();
+    OpenFileEntry oe = null;
+    synchronized (mOpenFiles) {
+      oe = mOpenFiles.get(fd);
+    }
+    if (oe == null) {
+      LOG.error("Cannot find fd for {} in table", path);
+      return -ErrorCodes.EBADFD();
+    }
+    if (oe.getOut() != null) {
+      try {
+        oe.getOut().flush();
+      } catch (IOException e) {
+        return -ErrorCodes.EIO();
+      }
+    } else {
+      LOG.debug("Not flushing: {} was not open for writing", path);
+    }
+    return 0;
+  }
+
+  /**
+   * Retrieves file attributes.
+   * @param path The path on the FS of the file
+   * @param stat FUSE data structure to fill with file attrs
+   * @return 0 on success, negative value on error
+   */
+  @Override
+  public int getattr(String path, FileStat stat) {
+    final TachyonURI turi = mPathResolverCache.getUnchecked(path);
+    LOG.trace("getattr({}) [Tachyon: {}]", path, turi);
+    try {
+      final TachyonFile tf = mTFS.openIfExists(turi);
+      if (tf == null) {
+        return -ErrorCodes.ENOENT();
+      }
+      final FileInfo fi = mTFS.getInfo(tf);
+      stat.st_size.set(fi.getLength());
+
+      final long ctime = fi.getLastModificationTimeMs();
+      final long ctime_sec = fi.getLastModificationTimeMs() / 1000;
+      //keeps only the "residual" nanoseconds not caputred in
+      // citme_sec
+      final long ctime_nsec = (fi.getLastModificationTimeMs() % 1000) * 1000;
+      stat.st_ctim.tv_sec.set(ctime_sec);
+      stat.st_ctim.tv_nsec.set(ctime_nsec);
+      stat.st_mtim.tv_sec.set(ctime_sec);
+      stat.st_mtim.tv_nsec.set(ctime_nsec);
+
+      // TODO(andreareale): understand how to map FileInfo#getUserName()
+      // and FileInfo#getGroupName() to UIDs and GIDs of the node
+      // where tachyon-fuse is mounted.
+      // While this is not done, just use uid and gid of the user
+      // running tachyon-fuse.
+      stat.st_uid.set(UID_AND_GID[0]);
```

Comment:
@andreareale , thanks for the detailed explanation. I misunderstood what you meant by mapping, and I don't think there's mapping between tachyon users/groups and ones on local machines.

I don't have good answers to those question, and I think it indeed is hard problem, or even it just doesn't make sense to have such mapping. But just curious, do you know if any HDFS FUSE connectors support such mapping?